### ISSUE TITLE
Ubuntu Server stuck on ./discuit -migrate

### ISSUE THREAD

**Issue by meneergroot** [discuitnet/discuit#33]
Hi again,

i tried installing discuit on my brand new ubuntu 22.04 server. 

Got to the last part ./discuit -migrate and ./discuit - serve

After the _Create a file named config.yaml in the root directory and copy the contents of config.default.yaml into it. And enter the required config parameters in config.yaml._

i did a _cp config.default.yaml config.yaml_ in the terminal, entered via nano config.yaml, added the server IP address, password for root. (I didnt throw away the config.default.yaml) and did ./build.sh

Started running, then after step 8. the command ./discuit -migrate i get _./discuit : no such file in directory_

What am i missing here.

**Comment by lydia-st** [comment]
What was the output of `./build.sh`?

**Comment by meneergroot** [comment]
If i put ./build.sh into the terminal now it states:
_core/community.go:10:2: package slices is not in GOROOT (usr/lib/go-1.18/src/slices)_

**Comment by lydia-st** [comment]
Ah that's not fun

If you run `echo $GOROOT $GOPATH`, what does it output? 
And also might be worth asking, how did you install Go?

**Comment by PendaGTP** [comment]
@Codycody31 Yup that's right, related to: https://github.com/discuitnet/discuit/pull/30#issue-2138407255

**Comment by meneergroot** [comment]
> Ah that's not fun
> 
> If you run `echo $GOROOT $GOPATH`, what does it output? And also might be worth asking, how did you install Go?

root/go

Did a whole server reset and started everything from scratch again (fourth time). Everytime i get stuck on step 8 ./discuit -migrate

output i get is _-bash: ./discuit: No such file or directory_

**Comment by Codycody31** [comment]
Try these. The main issue for you still appears to be that the discuit executable is still not being created

- Check the current directory to ensure the discuit file was created
- After running `./build.sh` do you still get the same error regarding slices
- attempt running `go build -o discuit .` directly instead of through `./build.sh`

**Comment by meneergroot** [comment]
> Try these. The main issue for you still appears to be that the discuit executable is still not being created
> 
> * Check the current directory to ensure the discuit file was created
> * After running `./build.sh` do you still get the same error regarding slices
> * attempt running `go build -o discuit .` directly instead of through `./build.sh`

The slices error isnt there. 
go build -o discuit . in the discuit directory gives:
<img width="1444" alt="Screenshot 2024-02-17 at 19 33 18" src="https://github.com/discuitnet/discuit/assets/53813521/9b8a860c-7b78-4c8b-8572-5a7aefcd3ba9">

**Comment by Codycody31** [comment]
see if you can install the package via: `apt-get install pkg-config`. That might fix it?

**Comment by meneergroot** [comment]
It's getting worse by the minute so it seems. 
The ./build.sh threw out:
<img width="1449" alt="Screenshot 2024-02-17 at 21 25 38" src="https://github.com/discuitnet/discuit/assets/53813521/5c2460c8-ed1b-4065-8ee5-b0e2095b2727">

**Comment by PendaGTP** [comment]
@meneergroot An error with `pkg-config` can be related to `libvips`. Do you have `libvips-dev` package installed on your system? (This will not resolve your other new issue with webpack.)

**Comment by meneergroot** [comment]
> @meneergroot An error with `pkg-config` can be related to `libvips`. Do you have `libvips-dev` package installed on your system? (This will not resolve your other new issue with webpack.)

yes
<img width="1166" alt="Screenshot 2024-02-17 at 21 32 47" src="https://github.com/discuitnet/discuit/assets/53813521/6bad8895-077a-497a-84d3-f688e39b64bb">

**Comment by PendaGTP** [comment]
Could you please provide the versions of Node.js and npm installed on your system?

**Comment by meneergroot** [comment]
Had a Ubuntu 23.10 container on my macbook and that worked after long thinkering. Now i Upgraded my Ubuntu server from 22.04 tot 23.10 and ./build.sh now works.

But.

./discuit -migrate gets a dial tcp 194.164.xxx.xxx:3306: connect : connection refused. 

(_getting a step further, baby steps but steps none the less_)

**Comment by Codycody31** [comment]
huh well that's better, it seems your DB server is unreachable.

**Comment by meneergroot** [comment]
> huh well that's better, it seems your DB server is unreachable.

That took me a while as a non-backender to find it. 
The Mariadb isnt bind to the IP address of my server 194.164.xxx.xxx but it uses the local 127.0.0.0.1 
I was to quick filling in my server address in the config.yaml

![im-sorry-i-fudged-up-guys](https://github.com/discuitnet/discuit/assets/53813521/aa34508b-f0d5-4604-b4d6-3e209465e46f)
